### PR TITLE
Lecture notes for Destructing Arrays

### PR DESCRIPTION
**Pull request by cchiw**
#274

### CODE REVIEW

File: src/pattern-matching/destructuring-arrays.md [hunk 1]
```diff
@@ -5,3 +5,10 @@ You can destructure arrays, tuples, and slices by matching on their elements:
 ```rust,editable
 {{#include ../../third_party/rust-by-example/destructuring-arrays.rs}}
 ```
+
+<details>
+  
+Key points:
+* Continuing our journey in pattern matching, this slide just shows another type that can destructured with pattern matching.
```
Comment (mgeisler) [inline src/pattern-matching/destructuring-arrays.md:12]: Let's take this line out: it's more narrative than a "key point" for the instructor.

File: src/pattern-matching/destructuring-arrays.md [hunk 2]
```diff
@@ -5,3 +5,10 @@ You can destructure arrays, tuples, and slices by matching on their elements:
 ```rust,editable
 {{#include ../../third_party/rust-by-example/destructuring-arrays.rs}}
 ```
+
+<details>
+  
+Key points:
+* Continuing our journey in pattern matching, this slide just shows another type that can destructured with pattern matching. 
+* This is a another great time to do some live coding. You can add/remove values from the array, compare the syntax `..` and `_`, and share any errors you get.
```
Comment (mgeisler) [inline src/pattern-matching/destructuring-arrays.md:13]: Can you update this with precise instructions on what to do?

Please make sure that the notes _add_ to the slide. They should explain things to people who don't have an instructor around as well as tell the instructor what they should focus on.

**Review by mgeisler — CHANGES_REQUESTED** [review]
Let's make the notes

1. Expand on the content with extra background. Perhaps we can explain if using pattern matching is more/less expensive than using the equivalent `if` expressions?
2. Give precise tips for the instructor on what to do.

**Comment by cchiw** [comment]
updated, let me know what you think

**Comment by cchiw** [comment]
soft ping. @mgeisler it needs your approval since you opened the change request.

**Review by mgeisler — APPROVED** [review]
Sure, let's get this in.

**Pull request merged**